crates/chainql-core/src/client/live.rs

```diff
@@ -229,15 +240,21 @@ impl LiveClient {
 			let key_str = format!("0x{}", hex::encode(key));
 			list.push(key_str);
 		}
-		eprintln!("preloading {} keys", list.len());
+
+		let keys_count = list.len();
+		let left = left - keys_count;
+
 		let handle = Handle::current();
 		let value = handle.block_on(
 			self.real
 				.query_storage(list, Some(self.block.as_str().to_owned())),
 		)?;
 		if value.is_empty() {
-			return Ok(());
+			return Ok(0);
 		}
+
+		info!(target: LOG_TARGET, "preloaded {keys_count} keys ({left} left)");
```

Comment:
If we really want to have nice output, we can use tracing spans here, then we can just use https://github.com/emersonford/tracing-indicatif for visualising span progress.

Leave it as-is for now.